packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -170,11 +170,8 @@ class ImageCache {
       // Images that fail to load don't contribute to cache size.
       final int imageSize = info?.image == null ? 0 : info.image.height * info.image.width * 4;
       final _CachedImage image = _CachedImage(result, imageSize);
-      // If the image is bigger than the maximum cache size, and the cache size
-      // is not zero, then increase the cache size to the size of the image plus
-      // some change.
       if (maximumSizeBytes > 0 && imageSize > maximumSizeBytes) {
-        _maximumSizeBytes = imageSize + 1000;
+        throw 'Attempting to cache an image whose size exceeds ImageCache.maximumSizeBytes: $maximumSizeBytes';
```

Comment:
This is a breaking change right?

Reply:
Nope, the exception is gobbled up here: https://github.com/flutter/flutter/blob/e60c0f68528d95d80783be58ba8b58cf49bce832/packages/flutter/lib/src/painting/image_stream.dart#L409:L409

Reply:
It's redirected to `FlutterError.reportError`, which will appear to users as if an exception is being thrown.

Reply:
(just not one they can easily catch with a `try .. catch`)

Reply:
That is my intention, no good?  Also, the test asserts that it behaves the same way btw.

Reply:
We do something similar when NetworkImage throws an exception.  It reports the error to the user by-way of exception.  But there is no meaningful way to catch it.

Reply:
You can catch it by setting https://api.flutter.dev/flutter/foundation/FlutterError/onError.html

And if you do nothing, you will now start seeing errors in situations where you didn't see errors before.  This is a breaking change.

Reply:
Adding some comments to the issue for this.